packages/client/src/lib.js

```diff
@@ -148,11 +149,15 @@ class Web3Storage {
               })
               const res = await request.json()
 
-              if (request.ok) {
-                return res.cid
-              } else {
+              if (!request.ok) {
                 throw new Error(res.message)
               }
+
+              if (res.cid !== carRoot) {
+                throw new Error(`Service ${endpoint} is compromised and CID received is not the same as generated by the client`)
```

Comment:
Wow haha I'd maybe tone this down a little so nobody panics. Maybe something more like "root CID mismatch, expected: x, received: y"

Reply:
```suggestion
                throw new Error(`root CID mismatch, expected: ${carRoot}, received: ${res.cid}`)
```